homeassistant/components/ping/binary_sensor.py

```diff
@@ -102,15 +113,43 @@ async def async_update(self) -> None:
 
 
 class PingData:
-    """The Class for handling the data retrieval."""
+    """The base class for handling the data retrieval."""
 
-    def __init__(self, host, count) -> None:
+    def __init__(self, hass, host, count) -> None:
         """Initialize the data object."""
+        self.hass = hass
         self._ip_address = host
         self._count = count
         self.data = {}
         self.available = False
 
+
+class PingDataICMPLib(PingData):
+    """The Class for handling the data retrieval using icmplib."""
+
+    def ping(self):
+        """Send ICMP echo request and return details."""
+        return icmp_ping(self._ip_address, count=self._count)
+
+    async def async_update(self) -> None:
+        """Retrieve the latest details from the host."""
+        data = await self.hass.async_add_executor_job(self.ping)
+        self.data = {
+            "min": data.min_rtt,
+            "max": data.max_rtt,
+            "avg": data.avg_rtt,
+            "mdev": "",
+        }
+        self.available = data.is_alive
+
+
+class PingDataSubProcess(PingData):
+    """The Class for handling the data retrieval using the ping binary."""
+
+    def __init__(self, hass, host, count) -> None:
+        """Initialize the data object."""
+        super().__init__(hass, host, count)
+
```

Comment:
I think that you can skip this as it's the same.

```suggestion
```

Reply:
Almost, sadly it still needs to figure out the ping command.

Reply:
Oh ha I missed that other part. Thanks for the commit to show me 😛